crates/spfs/src/tracking/env.rs

```diff
@@ -19,13 +26,199 @@ pub const ENV_SPEC_SEPARATOR: &str = "+";
 /// Recognized as an empty environment spec with no items
 pub const ENV_SPEC_EMPTY: &str = "-";
 
+// For the config file that contains the extra bind mounts
+const SPFS_FILE_SUFFIX_YAML: &str = ".spfs.yaml";
+const DEFAULT_LIVE_LAYER_FILENAME: &str = "layer.spfs.yaml";
+
+// For preventing recursive loops when reading list of layers spfs
+// spec files from the command line
+static SEEN_SPEC_FILES: Lazy<std::sync::Mutex<HashSet<PathBuf>>> =
+    Lazy::new(|| std::sync::Mutex::new(HashSet::new()));
```

Comment:
It doesn't look like this ever gets cleared, right? We have some long-running processes and I worry that we might even simply try to load the same file twice which would cause this to error with no way to recover. It's something that I can refactor if we run into it, but for the future I think we should avoid this kind of global state unless it actually lives in the command line crate.

Reply:
Yes, it's not cleared. I've added a function to clear it, in case we need it. I did feel while working on this, the things in `tracking/env.rs` seemed to be partly closer to command line level (parsing), and partly lower level (tag/digest resolution).